docs/advanced/extensibleenums.md

```diff
@@ -7,216 +7,142 @@ Extensible Enums are an enhancement of specific Vanilla enums to allow new entri
+```toml
+# In neoforge.mods.toml:
+[[mods]]
+## The file is relative to the output directory of the resources, or the root path inside the jar when compiled
+## The 'resources' directory represents the root output directory of the resources
+enumExtensions="META-INF/enumextensions.json"
 ```
 
-## Contributing to NeoForge
-
-To add a new extensible enum to NeoForge, there are at least two required things to do.
-
-First, have the enum implement `IExtensibleEnum` to mark that this enum should be transformed via the `RuntimeEnumExtender`. Second, add a `create` method for every constructor within the enum. All create methods should start with a `String` parameter representing the name of the enum constant. The `create` methods should throw an `IllegalStateException`.
-
-:::note
-The `create` method(s) will be transformed at runtime, so unless the enum was not detectable, the `create` method will create a new enum entry.
-:::
-
-```java
-// This is an example, not an actual enum within Vanilla
-public enum ExampleEnum implements net.neoforged.neoforge.common.IExtensibleEnum {
-    // VALUE_1 represents the name parameter here
-    VALUE_1(0, "value_1", false),
-    VALUE_2(1, "value_2", true),
-    VALUE_3(2, "value_3");
-
-    ExampleEnum(int arg1, String arg2, boolean arg3) {
-        // ...
-    }
-
-    ExampleEnum(int arg1, String arg2) {
-        this(arg1, arg2, false);
+The definition of the entry consists of the target enum's class name, the new field's name (must be prefixed with the mod ID), the descriptor of the constructor to use for constructing the entry and the parameters to be passed to said constructor.
+
+```json5
+{
+  "entries": [
+    {
+      // The enum class the entry should be added to
+      "enum": "net/minecraft/world/item/ItemDisplayContext",
+      // The field name of the new entry, must be prefixed with the mod ID
+      "name": "EXAMPLEMOD_STANDING",
+      // The constructor to be used
+      "constructor": "(ILjava/lang/String;Ljava/lang/String;)V",
+      // Constant parameters provided directly.
+      "parameters": [ -1, "examplemod:standing", null ]
+    },
+    {
+      "enum": "net/minecraft/world/item/Rarity",
+      "name": "EXAMPLEMOD_CUSTOM",
+      "constructor": "(ILjava/lang/String;Ljava/util/function/UnaryOperator;)V",
+      // The parameters to be used, provided as a reference to an EnumProxy<Rarity> field in the given class
+      "parameters": {
+        "class": "example/examplemod/MyEnumParams",
+        "field": "CUSTOM_RARITY_ENUM_PROXY"
+      }
+    },
+    {
+      "enum": "net/minecraft/world/damagesource/DamageEffects",
+      "name": "EXAMPLEMOD_TEST",
+      "constructor": "(Ljava/lang/String;Ljava/util/function/Supplier;)V",
+      // The parameters to be used, provided as a reference to a method in the given class
+      "parameters": {
+        "class": "example/examplemod/MyEnumParams",
+        "field": "getTestDamageEffectsParameter"
```

Comment:
Without looking at the system, I'm going to assume this is incorrect because the key is "field", and this is referencing a method.

Reply:
Indeed, copy-paste screwed me over once again